File: урок2/HWmedian.py
n = int(input())
arr = list(map(int, input().split()))
arr.sort()
ans = []
left = n // 2 - 1
right = n // 2
while len(ans) < n:
    if(n - len(ans)) % 2 == 1:
        ans.append(arr[right])
        right += 1
    else:
        ans.append(arr[left])
        left -= 1
    print(*ans)

File: урок2/HWmedian.js
function medians(arr) {
    const n = arr.length;
    arr.sort((a, b) => a - b);
    const ans = [];
    let left = Math.floor(n / 2 - 1);
    let right = Math.floor(n / 2);

    while (ans.length < n) {
        if ((n - ans.length) % 2 === 1) {
            ans.push(arr[right]);
            right++;
        } else {
            ans.push(arr[left]);
            left--;
        }
    }
    return ans;
}

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

let input = [];
readline.on('line', (line) => {
    input.push(line);
    if (input.length === 2) {
        readline.close();
    }
}).on('close', () => {
    const n = Number(input[0]);
    const arr = input[1].split(' ').map(Number).slice(0, n);
    const result = medians(arr); 
    console.log(result.join(' ')); 
});
